### PR TITLE
Expose file logging jni develop

### PR DESCRIPTION
**Pull request by niyatim23**
*Issue #, if available:*

*Description of changes:*
Set the callback for logPrintFn to a Java method

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/JNI/com/amazonaws/kinesis/video/producer/jni/KinesisVideoClientWrapper.cpp [hunk 1]
```diff
@@ -1011,7 +1016,7 @@ BOOL KinesisVideoClientWrapper::setCallbacks(JNIEnv* env, jobject thiz)
     // as the signature of the function does not have "custom_data"
     // to properly map to the client object.
     // We will use the default logger for Java.
-    mClientCallbacks.logPrintFn = NULL;
+    // mClientCallbacks.logPrintFn = NULL;
```
Comment (unicornss) [inline src/JNI/com/amazonaws/kinesis/video/producer/jni/KinesisVideoClientWrapper.cpp:1019]: /nit can we remove if not in use?

File: src/JNI/include/com/amazonaws/kinesis/video/producer/jni/KinesisVideoClientWrapper.h [hunk 2]
```diff
@@ -54,8 +54,8 @@
 class KinesisVideoClientWrapper
 {
     CLIENT_HANDLE mClientHandle;
-    JavaVM *mJvm;
-    jobject mGlobalJniObjRef;
+    static JavaVM *mJvm;
```
Comment (unicornss) [inline src/JNI/include/com/amazonaws/kinesis/video/producer/jni/KinesisVideoClientWrapper.h:57]: should we add a comment why we need to revise this scope?

File: CMake/Dependencies/libkvscproducer-CMakeLists.txt [hunk 3]
```diff
@@ -7,7 +7,7 @@ include(ExternalProject)
 # clone repo only
 ExternalProject_Add(libkvscproducer-download
 	GIT_REPOSITORY    https://github.com/awslabs/amazon-kinesis-video-streams-producer-c.git
-	GIT_TAG           69f68a7263efc6435c8dcf64ed163c9baa699325
+	GIT_TAG           4fc9d18d709e552f8d0b4f4f83a38c5dab726f86
```
Comment (unicornss) [inline CMake/Dependencies/libkvscproducer-CMakeLists.txt:10]: let us add the detail on how this works in conjunction with the log4j changes in the Java SDK

Comment (niyatim23) [reply CMake/Dependencies/libkvscproducer-CMakeLists.txt:10]: This git-tag is from the version of producer c with 3.6.3

File: src/JNI/com/amazonaws/kinesis/video/producer/jni/KinesisVideoClientWrapper.cpp (hunk 1 above)
Comment (niyatim23) [reply src/JNI/com/amazonaws/kinesis/video/producer/jni/KinesisVideoClientWrapper.cpp]: Addressed in the most recent commit

File: src/JNI/include/com/amazonaws/kinesis/video/producer/jni/KinesisVideoClientWrapper.h (hunk 2 above)
Comment (niyatim23) [reply src/JNI/include/com/amazonaws/kinesis/video/producer/jni/KinesisVideoClientWrapper.h]: Addressed in the most recent commit

File: src/JNI/com/amazonaws/kinesis/video/producer/jni/KinesisVideoClientWrapper.cpp [hunk 4]
```diff
@@ -2613,3 +2619,73 @@ AUTH_INFO_TYPE KinesisVideoClientWrapper::authInfoTypeFromInt(UINT32 authInfoTyp
...
+    if (envState == JNI_EDETACHED) {
+        if (mJvm->AttachCurrentThread((PVOID*) &env, NULL) != 0) {
+            goto CleanUp;
+        }
+        detached = TRUE;
+    }
+    
+    va_list list;
+    va_start(list, fmt);
+    vsnprintf(buffer, MAX_LOG_MESSAGE_LENGTH, fmt, list);
```
Comment (hassanctech) [inline src/JNI/com/amazonaws/kinesis/video/producer/jni/KinesisVideoClientWrapper.cpp:2643]: looks like `vsnprintf` is not available on Windows we'll have to change this to not fail the windows build

Comment (hassanctech) [reply src/JNI/com/amazonaws/kinesis/video/producer/jni/KinesisVideoClientWrapper.cpp:2643]: nevermind... was looking at something a decade old

**Review by hassanctech — APPROVED** [review]
lgtm

**Pull request merged**